Mathlib/Topology/MetricSpace/HoelderNorm.lean

```diff
@@ -0,0 +1,226 @@
+/-
+Copyright (c) 2024 Kexing Ying. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kexing Ying
+-/
+import Mathlib.Topology.MetricSpace.Hoelder
+
+/-!
+# Hölder norm
+
+This file defines the Hölder (semi-)norm for Hölder functions alongside some basic properties.
+The `r`-Hölder norm of a function `f : X → Y` between two metric spaces is the least non-negative
+real number `C` for which `f` is `r`-Hölder continuous with constant `C`, i.e. it is the least `C`
+for which `WithHoelder C r f` is true.
+
+## Main definitions
+
+* `eHoelderNorm r f`: `r`-Hölder (semi-)norm in `ℝ≥0∞` of a function `f`.
+* `nnHoelderNorm r f`: `r`-Hölder (semi-)norm in `ℝ≥0` of a function `f`.
+* `MemHoelder r f`: Predicate for a function `f` being `r`-Hölder continuous.
+
+## Main results
+
+* `eHoelderNorm_eq_zero`: the Hölder norm of a function is zero if and only if it is constant.
+* `MemHoelder.hoelderWith`: The Hölder norm of a Hölder function `f` is a Hölder constant of `f`.
+
+## Tags
+
+Hölder norm, Hoeder norm
+
+-/
+
+variable {X Y Z : Type*}
+
+open Filter Set
+
+open NNReal ENNReal Topology
+
+section PseudoEMetricSpace
+
+variable [PseudoEMetricSpace X] [PseudoEMetricSpace Y] {r : ℝ≥0} {f : X → Y}
+
+/-- The `r`-Hölder (semi-)norm in `ℝ≥0∞` of a function `f` is the least non-negative real
+number `C` for which `f` is `r`-Hölder continuous with constant `C`. This is `∞` if no such
+non-negative real exists. -/
+noncomputable
+def eHoelderNorm (r : ℝ≥0) (f : X → Y) : ℝ≥0∞ := ⨅ (C) (_ : HoelderWith C r f), C
+
+/-- The `r`-Hölder (semi)norm in `ℝ≥0`. -/
+noncomputable
+def nnHoelderNorm (r : ℝ≥0) (f : X → Y) : ℝ≥0 := (eHoelderNorm r f).toNNReal
+
+/-- A function `f` is `MemHoelder r f` if it is Hölder continuous. Namely, `f` has a finite
+`r`-Hölder constant. This is equivalent to `f` having finite Hölder norm.
+c.f. `memHoelder_iff`. -/
+def MemHoelder (r : ℝ≥0) (f : X → Y) : Prop := ∃ C, HoelderWith C r f
+
+lemma HoelderWith.memHoelder {C : ℝ≥0} (hf : HoelderWith C r f) : MemHoelder r f := ⟨C, hf⟩
+
+@[simp] lemma eHoelderNorm_lt_top_iff : eHoelderNorm r f < ∞ ↔ MemHoelder r f := by
+  refine ⟨fun h => ?_,
+    fun hf => let ⟨C, hC⟩ := hf; iInf_lt_top.2 ⟨C, iInf_lt_top.2 ⟨hC, coe_lt_top⟩⟩⟩
+  simp_rw [eHoelderNorm, iInf_lt_top] at h
+  exact let ⟨C, hC, _⟩ := h; ⟨C, hC⟩
+
+lemma eHoelderNorm_ne_top_iff : eHoelderNorm r f ≠ ∞ ↔ MemHoelder r f := by
```

Comment:
```suggestion
lemma eHoelderNorm_ne_top : eHoelderNorm r f ≠ ∞ ↔ MemHoelder r f := by
```

Reply:
`eHoelderNorm_ne_top` already exists... Same with the next comment

Reply:
Where? All I see is `MemHoelder.eHoelderNorm_ne_top`

Reply:
Changing the name to `eHoelderNorm_ne_top` breaks the alias and I dont know how to fix it

Reply:
What do you mean by "It breaks the alias"? Does `protected alias` unbreak it?